ql/src/java/org/apache/hadoop/hive/ql/ddl/DDLUtils.java

```diff
@@ -227,4 +229,17 @@ public static void validateTableIsIceberg(org.apache.hadoop.hive.ql.metadata.Tab
           table.getFullTableName(), tableType));
     }
   }
+
+  public static boolean isIcebergTable(Table table) {
+    return Optional.ofNullable(table.getStorageHandler())
```

Comment:
can't we reuse logic from validateTableIsIceberg?

Reply:
We can't because table parameter `table_type` gets put into the table only in `HiveIcebergMetaHook.preCreateTable` which is called from Metastore client 
```
    HiveMetaHook hook = getHook(tbl);
    if (hook != null) {
      hook.preCreateTable(tbl);
    }
```
Since we wanted to move the code that puts StatsSetupConst.DO_NOT_UPDATE_STATS into envContext on the caller side, the `table_type`parameter is not defined at that time.

Reply:
got it. in that case shouldn't it be something like this:
````
table.isNonNative() && HiveMetaHook.ICEBERG.equals(table.getStorageHandler().getTableType())
````

Reply:
Done